custom_components/ltss/models.py

```diff
@@ -27,9 +27,8 @@ class LTSS(Base):  # type: ignore
     """State change history."""
 
     __tablename__ = "ltss"
-    id = Column(BigInteger, primary_key=True, autoincrement=True)
     time = Column(DateTime(timezone=True), default=datetime.utcnow, primary_key=True)
-    entity_id = Column(String(255))
+    entity_id = Column(String(255), primary_key=True)
```

Comment:
@nwerter 
I am trying to understand whether the best course of action is to:
* Replace the old  primary key constraint (id, time) with a new one (time, entity_id), i.e. what you propose in this PR, or
* Remove all primary key constraints whatsoever?

Is it actually providing any benefits for this usecase? I am unsure. Do you have any input?

Reply:
Also, can we be sure that the composite primary key on (time, entity_id) will always be unique? I.e. that the same entity will never produce state change events with the same timestamp?

Reply:
@freol35241 
The main advantage I see of the primary key combination of (time, entity_id) is that we ensure that no duplicate entries can exist in the database. I'm not sure whether there is a scenario in which Home Assistant could send duplicate data to the database, but it doesn't hurt to have the primary keys to avoid this scenario.

The main reason to remove the id column is compression and with time and entity_id as primary keys, it's straigthforward to enable compression.

Unless you see another argument for removing the primary keys?

Reply:
In response to your second comment, the time data is stored in microseconds and I don't think an entity should produce two different states changes at the exact same timestamp. So even if there would be multiple state changes in quick succession they should always have a unique timestamp.

Reply:
Thanks for your input.

I think I agree, lets go with your proposal. In general though, it seems that a primary key might not be necessary at all (see for example https://stackoverflow.com/questions/66032256/how-to-create-timescaledb-hypertable-with-time-partitioning-on-non-unique-timest)